### ISSUE TITLE
Landsat Support

### ISSUE THREAD

**Issue by lukasValentin** [EOA-team/eodal#66]
This issue is a tracker for the progress on the full integration of Landsat into EOdal. The goal is to provide users with a similar smooth experience as with Sentinel-2 including the usage of the `Mapper` class. We focus on Collection-2 products (1982 to present) but also an integration of Collection-1 is feasible.

Thanks to the modular and sensor-agnostic design of the `core` and `mapper` submodule in EOdal, only few coding efforts are necessary.

- [x] define a class `Landsat` in `eodal.core.sensors` that inherits from the `RasterCollection` class
- [x] add a class-method `from_usgs` that allows to read Landsat data from GeoTiff files issued by [USGS](https://earthexplorer.usgs.gov/)
- [x] add a method to conveniently mask out clouds and other artifacts
- [x] ensure Landsat data can be fetched from STAC from Microsoft Planetary Computer
- [x] test the integration with the `Mapper` class

**Comment by lukasValentin** [comment]
While working on the new Landsat class, I realized that the [SpectralIndices](https://github.com/EOA-team/eodal/blob/master/eodal/core/spectral_indices.py) class which is called in the back-end whenever you execute the `RasterCollection.calc_si` method currently is too static and lacks the possibility to use custom color names.

This became evident as the near-infrared and shortwave-infrared bands in the Landsat Collection-2 tier are named slightly differently than their Sentinel-2 equivalents: In detail, the `nir_1` band is called `nir08`, `swir_1` `swir16` and `swir_2` `swir22`. The choice to call the NIR and SWIR bands `_1`, `_2`, etc. (instead of including the wavelength as the Landsat collection does) was a very early design decision made in EOdal and changing this would mean MAJOR changes required throughout the library.

Thus, I decided instead to make the [SpectralIndices](https://github.com/EOA-team/eodal/blob/master/eodal/core/spectral_indices.py) more flexible by adding a proper class constructor (so far not available) which takes an optional `band_mapping` dictionary as argument which is empty by default. This not only allows to map the Landsat color names into EOdal color names but also leaves (advanced) users of EOdal the choice to use a different spectral band for spectral index calculation.

For instance, if you are about to calculate the NDVI from Sentinel-2, EOdal by default uses Band 8 to do so. Band 8 comes at 10 m spatial resolution but its spectral resolution is coarse so some users might find Band 8A more suited which comes at a higher spectral resolution but at 20 m spatial resolution (thus, spatial resampling will be required). With the changes to be introduced in `v0.2.2` of EOdal, changing the bands used for SI calculation becomes fairly easy:

```python
from eodal.core.sensors import Sentinel2
# s2 must be a Sentinel2 (RasterCollection) instance
assert isinstance(s2, Sentinel2)

# calculate default NDVI using band 8 (nir_1)
ndvi_b08 = s2.calc_si('ndvi')

# calculate the NDVI using band 8A instead
# note: this will fail in case Band 8A is still in 20 m spatial resolution!
ndvi_b8a = s2.calc_si('ndvi', band_mapping={'nir_1': 'nir_2'})

```

This way, we avoid changing to many things in EOdal and leave a lot of freedom to the (advanced) user.

For Landsat, to conclude, the mapping looks as follows:

```python
band_mapping = {
    'nir_1': 'nir08',
    'swir_1': 'swir16',
    'swir_2': 'swir22'
}
```

**Comment by lukasValentin** [comment]
closed with #68
